recipes/cppitertools/all/conanfile.py

```diff
@@ -0,0 +1,59 @@
+import os
+from conans import ConanFile, tools
+from conans.errors import ConanInvalidConfiguration
+
+required_conan_version = ">=1.28.0"
+
+
+class CppItertoolsConan(ConanFile):
+    name = "cppitertools"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/ryanhaining/cppitertools"
+    description = "Implementation of python itertools and builtin iteration functions for C++17"
+    topics = ("cpp17", "iter", "itertools")
+    license = "BSD-2-Clause"
+    no_copy_source = True
+
+    settings = "os", "arch", "compiler", "build_type"
+    options = {'zip_longest': [True, False]}
+    default_options = {'zip_longest': False}
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = self.name + "-" + self.version
+        os.rename(extracted_dir, self._source_subfolder)
+
+    def validate(self):
+        # Requires C++17
+        minimal_version = {
+            "Visual Studio": "16",
+            "gcc": "7.3",
+            "clang": "6.0",
+            "apple-clang": "10.0"
```

Comment:
According to the README, tests pass at least for:
- Visual Studio 2017 and 2019
- gcc >= 7
- clang >= 5

https://github.com/ryanhaining/cppitertools#build-and-test-status